package onlineproto;

import "common.proto";

enum active_status_t
{
    ACTIVE_STATUS_OK = 0; // 登录成功
    ACTIVE_STATUS_WRONG_ACTIVE_CODE_TOO_MUCH = 1; // 激活码错误次数过多
    ACTIVE_STATUS_WRONG_VERIFY_CODE = 4; // 验证码错误
    ACTIVE_STATUS_ACCOUNT_UNSAFE = 5; // 账号存在异常，需要输入验证码
}

enum login_status_t
{
    LOGIN_STATUS_LOGIN_OK = 0;//登陆成功
    LOGIN_STATUS_WRONG_PASSWD_TOO_MUCH = 1; //密码错误次数过多
    LOGIN_STATUS_ACCOUNT_UNSAFE = 2; //账号异常
    LOGIN_STATUS_WRONG_VERIFY_CODE = 3; //验证码错误
    LOGIN_STATUS_LOGIN_IN_DIFF_CITY = 4; //异地登录
    LOGIN_STATUS_LOGIN_IN_DIFF_CITY_TOO_MUCH = 5; //异地登录.
}

//账户冻结原因
enum act_frozen_type_t {
    ACT_FROZEN_TYPE_USE_PLUGIN = 1; //使用外挂
    ACT_FROZEN_TYPE_DIRTY_WORDS = 2; //黄赌毒
    ACT_FROZEN_TYPE_MALBEHAVIOR = 3; //游戏内恶意行为(bug利用)

    ACT_FROZEN_TYPE_OTHER = 1000; //其他理由
}

//0x0001 --登陆账号
message cs_0x0001_login
{
    optional uint32 uid = 1;//米米号
    optional string email = 2;//邮箱
    optional string nick = 3;//昵称
    required string passwd = 4;//密码 

    optional bytes verify_image_session = 5; //
    optional bytes verify_code = 6; //验证码
    optional string tad = 7;//广告渠道信息
}

message sc_0x0001_login
{
    required login_status_t status = 1;//enum login_status_t
    optional string session = 2;//登陆会话

    optional commonproto.player_base_info_t base_info = 3;//玩家基本信息
    optional commonproto.equip_list_t equips = 4; //身上的装备
    optional commonproto.pet_info_t pet = 5; //展示的精灵

    //验证码
    optional bytes verify_image_session = 6;
    optional bytes verify_image = 7;

    //异地登录信息
    optional uint32 last_login_ip = 8;
    optional uint32 last_login_time = 9;
    optional string last_login_city = 10;
    optional string current_login_city = 11;

    //登录成功后发送
    optional bool is_active = 12;
}

// 0x0002 如果是创建账号 可能会先请求一个随机名字
message cs_0x0002_require_random_nick
{
	required uint32 sex = 1;	// 0 男 1 女
}
message sc_0x0002_require_random_nick
{
	required string nick = 1;
}

//0x0003 --创建角色
message cs_0x0003_create_role
{
    required uint32 prof = 1; // 职业
    required uint32 sex = 2; //性别
    required string nick = 3;//昵称
    required string session = 4;//验证session
    optional string tad = 5; //广告渠道信息
}
message sc_0x0003_create_role { }


//0x0004 --拉取服务器列表(默认推荐,否则拉取全部)
message cs_0x0004_get_svr_list
{
    required string session = 1; //还是需要用session验证合法性
    optional bool svr_recommend = 2 [default = true]; //是否需要服务器推荐
}

message sc_0x0004_get_svr_list
{
    repeated commonproto.svr_info_t online_svrs = 1;
    optional uint32 pre_online_id = 2; //用户上一次登录过的online_id(可能在上面的列表中)
}

// 刷新一张验证码图片
message cs_0x0005_get_verify_image { }
message sc_0x0005_get_verify_image 
{
	required bytes verify_image_session = 1; // 验证码图片session
    required bytes verify_image = 2; // 验证码图片PNG格式
}

// session登录
message cs_0x0006_session_login
{
    required uint32 userid = 1; // 代签的米米号
    required uint32 from_gameid = 2; // 从哪个平台代签过来
    required string session = 3; // 代签session
	optional string tad = 4; // 推广渠道标识
}

message sc_0x0006_session_login
{
    required login_status_t status = 1;//enum login_status_t
    optional string session = 2;//登陆会话

    required commonproto.player_base_info_t base_info = 3;//玩家基本信息
    optional commonproto.equip_list_t equips = 4; //身上的装备
    optional commonproto.pet_info_t pet = 5; //展示的精灵

    //异地登录信息
    optional uint32 last_login_ip = 8;
    optional uint32 last_login_time = 9;
    optional string last_login_city = 10;
    optional string current_login_city = 11;

    //登录成功后发送
    optional bool is_active = 12;
}

//封测玩家激活码激活账号
message cs_0x0007_active_user
{
    required string active_code = 1; // 激活码
    optional bytes verify_image_session = 2; // 验证码图片session
    optional bytes verify_code = 3; // 验证码
}
message sc_0x0007_active_user
{
    required active_status_t status = 1; // 激活状态
    optional bytes verify_image_session =2; // 验证码图片session
    optional bytes verify_image = 3; // 验证码图片PNG格式
}

//服务器提示账户被冻结
message cs_0x0008_noti_act_frozen {}
message sc_0x0008_noti_act_frozen {
    required act_frozen_type_t type = 1;
    optional uint32 data = 3;
}

// 客户端统计项
message cs_0x0009_cli_stat
{
    required string stid = 1;
	required string sstid = 2;
    optional string dir = 3 [default = ""] ; // 默认放入统计平台活动目录下
}
message sc_0x0009_cli_stat {}
